File: cpp/akka_frnd.cpp
#include<bits/stdc++.h>
using namespace std;

/*void printARR(int ar[], int n){
    for (int i = 0; i < n; i++)
    {
        cout << ar[i] << ' ';
    }
    cout << endl;
    
}*/

int main(){

    int n;
    cin >> n;
    int arr[n];

    for (int i = 0; i < n; i++){
        cin >> arr[i];
    }

    int cnt = 0;

    for(int i=0; i<n; i++){

        if(arr[i] == 1){
            continue;
        }else{
            for(int j=i; j<n-1; j++){
                arr[j] = !arr[j];
            }

            cnt++;
        }

    }

    cout << cnt;

    return 0;

}

File: py/akka_frnd.py

def play_the_game(arr):
    n = len(arr)
    cnt=0

    while (arr.count(0)!=n):
        for i in range(n):
            if(arr[i]%2 == 1):
                arr[i] -= 1
                cnt+=1

        for i in range(n):
            arr[i] = arr[i]//2
            cnt += 1/n

    return cnt-1 



arr = input().split(" ")
arr = list(map(int, arr))
n = len(arr)
sum = round(play_the_game(arr), 0)
print(int(sum))


